### PR TITLE
SA-953 - Add Week and Month precision labels for Metrics Summary Chart

### PR DESCRIPTION
**Pull request by kchu1337**
### What Changed
 - Added Options for month and week labels for the tooltip.
 - Added Monthly display for the x-axis ticks.

### How To Test
 - Go to /reports/summary. Change the date ranges to see hourly, <hourly, daily, weekly, and monthly precision.

### To Do
- [ ] UX/TPM approval

### CODE REVIEW

File: src/helpers/chart.js [hunk 1]
```diff
@@ -19,17 +19,53 @@ function getDayLines(data, precision = 'day') {
...
 
-const getTooltipLabelFormatter = _.memoize((precisionType) => {
-  let labelFormat = 'MMMM Do';
-  if (precisionType === 'hours') {
-    labelFormat = 'MMM Do [at] LT';
+const getTooltipLabelFormatter = _.memoize(precisionType => {
+  const format = labelFormat => label => moment(label).format(labelFormat);
+  switch (precisionType) {
+    case 'hour':
+    case 'hours':
```
Comment (shrirupa) [inline src/helpers/chart.js]: does the api return both hour/hours, week/weeks as precision type ?

Comment (kchu1337) [reply src/helpers/chart.js]: I think it's just the singular. But I have both just in case.

Comment (shrirupa) [reply src/helpers/chart.js]: can we remove the plurals, if api is returning only singulars?

File: src/pages/reports/summary/components/ChartGroup.js [hunk 2]
```diff
@@ -10,53 +10,54 @@ function getUniqueUnits(metrics) {
...
       x: ts,
       stroke: '#bbb',
-      strokeWidth: 2
+      strokeWidth: 2,
     }));
   }
 
   render() {
-    const { chartData = [], metrics, chartLoading, precision, yScale } = this.props;
+    const { chartData = [], metrics, chartLoading, precision, yScale, to } = this.props;
```
Comment (kchu1337) [inline src/pages/reports/summary/components/ChartGroup.js]: I added this

File: src/pages/reports/summary/components/ChartGroup.js [hunk 3]
```diff
@@ -10,53 +10,54 @@ function getUniqueUnits(metrics) {
...
 
     if (!chartData.length || !metrics) {
       return null;
     }
 
-    const formatters = getLineChartFormatters(precision);
+    const formatters = getLineChartFormatters(precision, to);
```
Comment (kchu1337) [inline src/pages/reports/summary/components/ChartGroup.js]: I added this

File: src/pages/reports/summary/SummaryPage.js [hunk 4]
```diff
@@ -27,41 +27,48 @@ export class SummaryReportPage extends Component {
...
-  handleScaleClick = (scale) => {
+  handleScaleClick = scale => {
     this.setState({ scale });
-  }
+  };
 
   render() {
-    const { chart, summarySearchOptions, enhancementsEnabled } = this.props;
+    const { chart, summarySearchOptions = {}, enhancementsEnabled } = this.props;
+    const { to } = summarySearchOptions;
```
Comment (kchu1337) [inline src/pages/reports/summary/SummaryPage.js]: I added this

Comment (kchu1337) [inline src/pages/reports/summary/SummaryPage.js]: I added this

File: src/pages/reports/summary/SummaryPage.js [hunk 5]
```diff
@@ -70,7 +77,7 @@ export class SummaryReportPage extends Component {
               onTimeClick={this.handleTimeClick}
               onMetricsToggle={this.handleMetricsModal}
             />
-            <ChartGroup {...chart} yScale={scale} />
+            <ChartGroup {...chart} to={to} yScale={scale} />
```
Comment (kchu1337) [inline src/pages/reports/summary/SummaryPage.js]: I added this

File: src/helpers/chart.js [hunk 6]
```diff
@@ -45,35 +41,42 @@ const getTooltipLabelFormatter = _.memoize(precisionType => {
...
-    case 'months':
       return format('MMMM YYYY');
 
     default:
       return format('MMMM Do');
   }
 });
 
-function getLineChartFormatters(precision) {
+const getWeekPrecisionFormatter = to => {
```
Comment (shrirupa) [inline src/helpers/chart.js]: should we memoize this function?

**Pull request merged**